src/main/java/com/example/demo/tournament/CSVcontroller.java

```diff
@@ -0,0 +1,51 @@
+package com.example.demo.tournament;
+
+import com.example.demo.model.Team;
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.CSVWriter;
+import com.opencsv.bean.StatefulBeanToCsv;
+import com.opencsv.bean.StatefulBeanToCsvBuilder;
+import com.opencsv.exceptions.CsvDataTypeMismatchException;
+import com.opencsv.exceptions.CsvRequiredFieldEmptyException;
+import org.springframework.stereotype.Controller;
+
+import java.io.*;
+import java.util.List;
+
+@Controller
```

Comment:
Although @Controller technically works here, semantically this is not a controller.  Controller is a part of MVC pattern (https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) and this component doesn't look like controller. Simply make it a @Component